pkg/alignchecker/alignchecker.go

```diff
@@ -143,3 +145,152 @@ func check(name string, toCheck []reflect.Type, structs map[string]structInfo, c
 
 	return nil
 }
+
+// ErrElf marks an error trying to parse ELF file
+type ErrElf struct {
+	msg string
+}
+
+func (e *ErrElf) Error() string {
+	return fmt.Sprintf("elf error: %s", e.msg)
+}
+
+// ErrSizeMismatch marks a size mismatch
+type ErrSizeMismatch struct {
+	Name             string
+	SizeBpf, SizeVal uint64
+}
+
+func (e *ErrSizeMismatch) Error() string {
+	return fmt.Sprintf("size mismatch for symbol %s: bpf size:%d val size:%d", e.Name, e.SizeBpf, e.SizeVal)
+}
+
+// ErrSym marks a generic error for a given symbol
+type ErrSym struct {
+	Name string
+	msg  string
+}
+
+func (e *ErrSym) Error() string {
+	return fmt.Sprintf("error for symbol %s: %s", e.Name, e.msg)
+}
+
+// ErrValueMismatch marks a value mismatch
+type ErrValueMismatch struct {
+	Name        string
+	ValBpf, Val reflect.Value
+}
+
+func (e *ErrValueMismatch) Error() string {
+	return fmt.Sprintf("value mismatch for symbol %s: bpf:%v val:%v", e.Name, e.ValBpf, e.Val)
+}
+
+// valueCheck reads ELF symbols from .rodata and compare the values against given toCheck values.
+// It returns  the first error it encouters, or nil if succesful.
+func valueCheck(pathToObj string, toCheck map[string]reflect.Value) error {
+	f, err := elf.Open(pathToObj)
+	if err != nil {
+		return fmt.Errorf("failed to open %s: %w", pathToObj, err)
+	}
+	defer f.Close()
+
+	elfSyms, err := f.Symbols()
+	if err != nil {
+		return &ErrElf{"no symbols"}
+	}
+
+	elfSymsMap := map[string]elf.Symbol{}
```

Comment:
Using the `make` builtin to create empty maps is more idiomatic and would also allow to specify the expected size:

```suggestion
	elfSymsMap := make(map[string]elf.Symbol, len(elfSyms))
```